alf/utils/common.py

```diff
@@ -259,3 +259,26 @@ def _transform_image(obs):
             return obs
 
     return tf.nest.map_structure(_transform_image, observation)
+
+
+@gin.configurable
+def reward_clipping(r, minmax=(-1, 1)):
+    """
+    Clamp immediate rewards to the range [`min`, `max`].
+
+    Can be used as a reward shaping function passed to an algorithm
+    (e.g. ActorCriticAlgorithm).
+    """
+    assert minmax[0] <= minmax[1], "range error"
+    return tf.clip_by_value(r, minmax[0], minmax[1])
+
+
+@gin.configurable
+def reward_scaling(r, scale=1):
```

Comment:
simply scaling the reward won't have any meaningful effect because the gradient is proportionally scaled.

Reply:
Depending on the init values of the value functions, the advantages or td errors are not linearly correlated to the reward